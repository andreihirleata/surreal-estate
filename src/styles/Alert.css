.Alert {
  text-align: center;
  font-weight: bold;
}

.alert-success {
  color: green;
  border: 2px solid green;
}

.alert-error {
  color: red;
  border: 2px solid red;
}
