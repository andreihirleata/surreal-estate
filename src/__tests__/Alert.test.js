import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  test("it renders error message", () => {
    const { asFragment } = render(<Alert message="error" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("it renders success message", () => {
    const { asFragment } = render(<Alert message="success" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("it does not render an error or success message if message is empty", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("displays an error message", () => {
    const { getByText } = render(<Alert message="Error!" />);

    expect(getByText(/error/i).textContent).toBe("Error!");
  });

  test("displays success message", () => {
    const { getByText } = render(<Alert message="Success!" success />);

    expect(getByText(/success/i).textContent).toBe("Success!");
  });
});
