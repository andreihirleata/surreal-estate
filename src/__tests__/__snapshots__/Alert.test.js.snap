// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert it does not render an error or success message if message is empty 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  />
</DocumentFragment>
`;

exports[`Alert it renders error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    error
  </div>
</DocumentFragment>
`;

exports[`Alert it renders success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    success
  </div>
</DocumentFragment>
`;
